File: flink-app/src/main/java/com/example/ctr/config/KafkaProperties.java
package com.example.ctr.config;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class KafkaProperties {

    @NotBlank
    private String bootstrapServers;

    @NotBlank
    private String groupId;

    private Topics topics = new Topics();

    @Getter
    @Setter
    public static class Topics {
        @NotBlank
        private String impression;
        @NotBlank
        private String click;
    }
}

File: flink-app/src/main/kotlin/com/example/ctr/config/KafkaProperties.kt
package com.example.ctr.config

import jakarta.validation.constraints.NotBlank

class KafkaProperties {
    @field:NotBlank
    var bootstrapServers: String = ""

    @field:NotBlank
    var groupId: String = ""

    var topics: Topics = Topics()

    class Topics {
        @field:NotBlank
        var impression: String = ""

        @field:NotBlank
        var click: String = ""
    }
}
